Feature/first step basic appservice (#5)
Azure-Samples/app-service-baseline-implementation: README.md

## Prerequisites

1. Ensure you have an [Azure Account](https://azure.microsoft.com/free/)

> 1. It is needed an user with [roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles) with enough permission to do role assigments. For example User Access Administrator or Owner.

Reviewer: ```suggestion
1. The deployment must be started by a user who has sufficient permissions to assign [roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles), such as a User Access Administrator or Owner.
```

     ```

1. Update the infra-as-code/parameters file

```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "baseName": {
      "value": ""
    },
    "sqlAdministratorLogin": {},
    "sqlAdministratorLoginPassword": {},
    "developmentEnvironment": {
      "value": true
    },
    "appGatewayListenerCertificate": {
      "value": "[base64 cert data from $APP_GATEWAY_LISTENER_CERTIFICATE_APPSERV_BASELINE]"
    }
  }
}
```

Note: Take into account that sql database enforce [password complexity](https://learn.microsoft.com/sql/relational-databases/security/password-policy?view=sql-server-ver16#password-complexity)

1. Run the following command to create a resource group and deploy the infrastructure. Make sure:

   - The location you choose [supports availability zones](https://learn.microsoft.com/azure/reliability/availability-zones-service-support)
   - The BASE_NAME contains only lowercase letters and is between 6 and 12 characters. All resources will be named given this basename.
   - You choose a valid resource group name

```bash
   LOCATION=westus3

>    BASE_NAME=<base-resource-name between 3 and 6 charcters>

Reviewer: ```suggestion
   BASE_NAME=<base-resource-name (between 3 and 6 characters)>

```

1. The deployed storage account does not allow public access, so you will need to temporarily allow access public access from your IP address.
1. You need to give your user permissions to upload a blob to the storage account.

First, we need to clone the [Simple Web App workload repository](https://github.com/Azure-Samples/app-service-sample-workload)

```bash
cd ..

> clone https://github.com/Azure-Samples/app-service-sample-workload.git

Reviewer: git clone(?)

```suggestion
git clone https://github.com/Azure-Samples/app-service-sample-workload.git

```

Run the following to:

- Allow public access from your IP address, g
- Give the logged in user permissions to upload a blob
- Create the `deploy` container
- Upload the zip file `./website/SimpleWebApp/SimpleWebApp.zip` to the `deploy` container
- Tell the web app to restart

```bash
CLIENT_IP_ADDRESS=<your-public-ip-address>

STORAGE_ACCOUNT_PREFIX=st
WEB_APP_PREFIX=app-

> NAME_OF_WEST_STORAGE_ACCOUNT="$STORAGE_ACCOUNT_PREFIX$BASE_NAME"

Reviewer: Can we rename this variable to remove WEST from the name? Maybe just NAME_OF_STORAGE_ACCOUNT

Author: done
